test/integration/gardenlet/vpaevictionrequirements/vpaevictionrequirements_test.go

```diff
@@ -0,0 +1,211 @@
+		}
+		maintenanceWindowAlreadyPassed = &gardencorev1beta1.MaintenanceTimeWindow{
+			Begin: time.Now().Add(-3 * time.Hour).Format("150405-0700"),
+			End:   time.Now().Add(-2 * time.Hour).Format("150405-0700"),
+		}
+
+		targetDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-deployment",
+				Namespace: testNamespace.Name,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "test"}},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "test"}},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{Name: "test-container", Image: "my-nonexisting-image"},
+						},
+					},
+				},
+			},
+		}
+
+		vpa = &vpaautoscalingv1.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-vpa",
+				Namespace: testNamespace.Name,
+				Labels: map[string]string{
+					testID: testRunID,
+					v1beta1constants.LabelVPAEvictionRequirementsController: v1beta1constants.EvictionRequirementManagedByController,
+				},
+			},
+			Spec: vpaautoscalingv1.VerticalPodAutoscalerSpec{
+				TargetRef:    &autoscalingv1.CrossVersionObjectReference{Name: targetDeployment.Name, APIVersion: "apps/v1", Kind: "Deployment"},
+				UpdatePolicy: &vpaautoscalingv1.PodUpdatePolicy{EvictionRequirements: []*vpaautoscalingv1.EvictionRequirement{}},
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		By("Create deployment")
+		Expect(testClient.Create(ctx, targetDeployment)).To(Succeed())
+
+		DeferCleanup(func() {
+			By("Delete deployment")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, targetDeployment))).To(Succeed())
+		})
+
+		By("Create VPA")
+		Expect(testClient.Create(ctx, vpa)).To(Succeed())
+
+		DeferCleanup(func() {
+			By("Delete VPA")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, vpa))).To(Succeed())
+		})
+	})
+
+	Context("VPA is labeled with downscale-in-maintenance-window-only", func() {
+		BeforeEach(func() {
+			metav1.SetMetaDataLabel(&vpa.ObjectMeta, v1beta1constants.LabelVPAEvictionRequirementDownscaleRestriction, v1beta1constants.EvictionRequirementInMaintenanceWindowOnly)
+		})
+
+		When("the Shoot is outside its maintenance window", func() {
+			BeforeEach(func() {
+				metav1.SetMetaDataAnnotation(&vpa.ObjectMeta, v1beta1constants.AnnotationShootMaintenanceWindow, maintenanceWindowAlreadyPassed.Begin+","+maintenanceWindowAlreadyPassed.End)
+			})
+
+			It("should add an EvictionRequirement denying scaling down and requeue for the beginning of the next window", func() {
+				Eventually(func(g Gomega) {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), vpa)).To(Succeed())
+					g.Expect(vpa.Spec).To(MatchFields(IgnoreExtras, Fields{
+						"UpdatePolicy": PointTo(MatchFields(IgnoreExtras, Fields{
+							"EvictionRequirements": ConsistOf(upscaleOnlyRequirement),
+						})),
+					}))
```

Comment:
Any reasons to use expectation with IgnoreExtras instead of
```suggestion
					g.Expect(vpa.Spec.UpdatePolicy.EvictionRequirements).To(ConsistOf(upscaleOnlyRequirement))
```

?

The same question applies to the other usages of similar expectations with IgnoreExtras.

Reply:
done.